File: easiness-core/src/main/java/com/easiness/core/entities/StockEntity.java
package com.easiness.core.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "stock")
public class StockEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    ProductEntity product;

    @Column(name = "cost")
    BigDecimal cost;

    @Column(name = "latest_price")
    BigDecimal latestPrice;

    @Column(name = "quantity")
    BigDecimal quantity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    UnitEntity unit;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "place_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    PlaceEntity place;

}

File: easiness-ui/app/server/entity/stock.entity.ts
import Big from "big.js";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn, Unique } from "typeorm";
import { BigColumn } from "../utils/decorators";
import { Base } from "./base.entity";
import { Place } from "./place.entity";
import { Product } from "./product.entity";
import { Unit } from "./unit.entity";

export const uniqueStockCols = ['productId', 'placeId']
@Entity({ name: 'stock' })
@Unique(uniqueStockCols)
export class Stock extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @ManyToOne(() => Product)
    product: Product
    @Column()
    productId: number

    @BigColumn()
    cost: Big;

    @BigColumn(true)
    latestPrice!: Big;

    @BigColumn()
    quantity: Big;

    @ManyToOne(() => Unit)
    unit: Unit
    @Column()
    unitId: number

    @ManyToOne(() => Place)
    place: Place
    @Column()
    placeId: number

    public getAltId() {
        return `${this.productId}_${this.placeId}`
    }
}